### PR TITLE
fix failing tests. The failures for rails v > 6.1

### PR DESCRIPTION
**Pull request by arjun-rajappa**
… test app was not being initialised. So Instana activator for action view was not being triggered properly.

### CODE REVIEW

**Review by GSVarsha — CHANGES_REQUESTED** [review]
Can you have a shorter PR title and put all the details in the description?

**Review by GSVarsha — DISMISSED** [review]
The [CI tests](https://app.circleci.com/pipelines/github/instana/ruby-sensor?branch=fix-failing-action-view-tests) for rails are failing for this PR.

File: test/instrumentation/rails_action_view_test.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def app
   def setup
     clear_all!
     @framework_version = Gem::Specification.find_by_name('rails').version
-    @supported_framework_version = @framework_version < Gem::Version.new('6.1')
+    @supported_framework_version = @framework_version < Gem::Version.new('7.2')
```
Comment (Ferenc-) [inline test/instrumentation/rails_action_view_test.rb:21]: I think now all these from line `20` to `31` should be removed, and also the calls to these lambdas.

**Review by Ferenc- — CHANGES_REQUESTED** [review]

File: lib/instana/activators/action_view.rb [hunk 2]
```diff
@@ -12,8 +12,11 @@ def instrument
         require 'instana/instrumentation/action_view'
 
         ::ActionView::PartialRenderer
-          .prepend(Instana::Instrumentation::ActionView)
-
+          .prepend(Instana::Instrumentation::ActionView::PartialRenderer)
+        if ::Rails::VERSION::STRING > "6.0" && defined?(::ActionView::CollectionRenderer)
```
Comment (Ferenc-) [inline lib/instana/activators/action_view.rb:16]: I think if `::ActionView::CollectionRenderer` is defined, then we should prepend, regardless of the Rails version, afterall we have to cover all and any Rails versions.

File: test/support/apps/action_view/config.ru [hunk 3]
```diff
@@ -11,41 +11,27 @@ class TestViewApplication < Rails::Application
   config.consider_all_requests_local = true
   config.secret_key_base = 'test_key'
   config.secret_token = 'test_token'
+  config.autoload_paths = []
+  ActiveSupport::Dependencies.autoload_paths = ActiveSupport::Dependencies.autoload_paths.dup
```
Comment (Ferenc-) [inline test/support/apps/action_view/config.ru:15]: Why do we need this line?

File: lib/instana/activators/action_view.rb (hunk 2 above)
Comment (arjun-rajappa) [reply lib/instana/activators/action_view.rb:16]: Yeah makes sense will remove rails version check!

File: test/support/apps/action_view/config.ru (hunk 3 above)
Comment (arjun-rajappa) [reply test/support/apps/action_view/config.ru:15]: ActiveSupport::Dependencies.autoload_paths are being set globally and is frozen in our test environment during initialisation of our ActionControllerTest. So these lines would unfreeze the array and helps initialise the ActionViewTest app properly. Tried turning of the autoload but did not work so i am forcefully setting these(Unfreezing).
There should be no issues as ideally in a real world rails there will be one application(In our test env we have multiple apps)

**Comment by arjun-rajappa** [comment]
> The [CI tests](https://app.circleci.com/pipelines/github/instana/ruby-sensor?branch=fix-failing-action-view-tests) for rails are failing for this PR.



> Can you have a shorter PR title and put all the details in the description?



> The [CI tests](https://app.circleci.com/pipelines/github/instana/ruby-sensor?branch=fix-failing-action-view-tests) for rails are failing for this PR.



> The [CI tests](https://app.circleci.com/pipelines/github/instana/ruby-sensor?branch=fix-failing-action-view-tests) for rails are failing for this PR.

All tests are passing

**Pull request merged**